packages/commcare/src/Adaptor.js

```diff
@@ -77,23 +78,62 @@ export function get(path, params = {}, callback = s => s) {
       params
     );
 
+    let { offset = 0, limit = 1000 } = resolvedParams;
+
+    let nextState = state;
+    let result;
+    let allowPagination = !isNaN(resolvedParams.offset);
+
     try {
-      const response = await request(
-        state.configuration,
-        `/a/${domain}/api/v0.5/${resolvedPath}`,
-        {
-          method: 'GET',
-          params: resolvedParams,
-          contentType: 'application/json',
+      let requestParams = {
+        ...resolvedParams,
+      };
+
+      do {
+        const response = await request(
+          state.configuration,
+          `/a/${domain}/api/v0.5/${resolvedPath}`,
+          {
+            method: 'GET',
+            params: requestParams,
+            contentType: 'application/json',
+          }
+        );
+
+        nextState = prepareNextState(state, response, callback);
+
+        if (response?.body?.meta?.next) {
+          if (!result) {
+            result = [];
+          }
+          offset = response?.body?.meta?.offset + response?.body?.meta?.limit;
+          limit = response?.body?.meta?.limit;
+
+          requestParams = {
+            ...requestParams,
+            offset,
+            limit,
+          };
+          result.push(...nextState.data);
+        } else {
+          if (result) {
```

Comment:
My spidey sense looks at this block and says it's too complicated and badly structured. I wouldn't mind rethinking how `result` is handled.

The clue is that `result.push` is repeated. And anyway over this six line block, all we're really doing is assigning the result